assets/src/js/admin/reports/components/list/index.js

```diff
@@ -1,6 +1,35 @@
-const List = (props) => {
+import { useEffect, createRef } from 'react'
+import './style.scss';
+
+const List = ({onScrollEnd, children}) => {
+
+    const list = createRef()
+
+    useEffect(() => {
+
+        function checkScroll (evt) {
+
+            const remaining = evt.target.scrollHeight - evt.target.scrollTop
+            const height = evt.target.offsetHeight
+
+            if (remaining <= height) {
+                onScrollEnd()
+            }
+
+        }
+        
+        list.current.addEventListener('scroll', checkScroll)
+    
+        return function cleanup () {
+            list.current.removeEventListener('scroll', checkScroll)
+        }
+
+    }, [])
```

Comment:
Since you're referring to the `onScrollEnd` prop you'll want to make it a dependency of `useEffect`.

Reply:
Done! onScrollEnd is now passed to useEffect

Reply:
Sorry, that's not what I meant. 😄 

Parameters aren't used with `useEffect` as it's just a generic callback. When I said "make it a dependency", I meant:
```js
useEffect(() => {
  // code stuff
}, [onScrollEnd]);
```